src/UI/QuickmenuView.re

```diff
@@ -0,0 +1,243 @@
+
+
+let createElement =
+    (
+      ~children as _,
+      ~font: Types.UiFont.t,
+      ~theme: Theme.t,
+      ~configuration: Configuration.t,
+      ~autofocus: bool=true,
+      ~state: Quickmenu.t,
+      ~placeholder: string="type here to search the menu",
+      ~onInput: (string, int) => unit = onInput,
+      ~onSelectedChange: int => unit = onSelectedChange,
+      ~onSelect: int => unit = onSelect,
+      (),
+    ) =>
+  component(hooks => {
+    let Quickmenu.{source, selected, text, cursorPosition, prefix} = state;
+
+    // TODO: Cam this be removed in favor of NotifyKeyPressed in MenuStoreConnector?
+    let handleKeyDown = (event: NodeEvents.keyEventParams) =>
+      switch (event) {
+      | {key: Revery.Key.KEY_DOWN, _} =>
+        GlobalContext.current().dispatch(MenuFocusNext)
+      | {key: Revery.Key.KEY_UP, _} =>
+        GlobalContext.current().dispatch(MenuFocusPrevious)
+      | _ => ()
+      };
+
+    let (items, jobProgress) =
+      switch (source) {
+        | Loading =>
+          ([||], 0.)
+
+        | Progress({ items, progress }) =>
+          (items, progress)
+
+        | Complete(items) =>
+          (items, 1.)
+      };
+
+    let loadingSpinner =
+      if (jobProgress == 0.) {
+        <AnimatedView duration=2.>
+          ...(opacity =>
+            <View style=Style.[height(40), width(Constants.menuWidth)]>
+              <Center>
+                <AnimatedView duration=(2. *. Float.pi) repeat=true>
+                  ...(t =>
+                    <View
+                      style=Style.[
+                        transform(
+                          Transform.[RotateY(Math.Angle.Radians(t *. 2.))],
+                        ),
+                      ]>
+                      <Opacity opacity>
+                        <Container
+                          width=10
+                          height=10
+                          color={theme.oniNormalModeBackground}
+                        />
+                      </Opacity>
+                    </View>
+                  )
+                </AnimatedView>
+              </Center>
+            </View>
+          )
+        </AnimatedView>
+      } else {
+        <Opacity opacity=0.3>
+          <View style=Style.[height(2), width(Constants.menuWidth)]>
+            <View
+              style=Style.[
+                height(2),
+                width(
+                  1
+                  + (
+                    int_of_float(float_of_int(Constants.menuWidth) *. jobProgress)
+                    - 1
+                  ),
+                ),
+                backgroundColor(theme.oniNormalModeBackground),
+              ]
+            />
+          </View>
+        </Opacity>
+      };
+
+    let renderItem = index => {
+      let item = items[index];
+
+      let labelView = {
+        let font = GlobalContext.current().state.uiFont.fontFile;
+        let normalStyle = Styles.label(~font, ~fg=theme.editorMenuForeground);
```

Comment:
One quirk about our text rendering right now (pre-Skia...) is that we actually need to know the background color to get full fidelity (gamma corrected font). If we add the background color - the text will render a bit crisper.

Once we have Skia integrated, it won't be necessary to do this anymore

Reply:
Ah, so that's why. I couldn't see any difference, but didn't look _that_ closely. Will fix!